src/main/java/com/minecolonies/inventory/InventoryField.java

```diff
@@ -0,0 +1,344 @@
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public boolean isUseableByPlayer(final EntityPlayer entityPlayer)
+    {
+        return true;//colony.getPermissions().hasPermission(entityPlayer, Permissions.Action.ACCESS_HUTS);
+    }
+
+    /**
+     *  Called when inventory is opened by a player.
+     * @param entityPlayer the player who opened the inventory.
+     */
+    @Override
+    public void openInventory(final EntityPlayer entityPlayer)
+    {
+        /*
+         * This may be filled in order to specify some custom handling.
+         */
+    }
+
+    /**
+     * Called after the inventory has been closed by a player.
+     * @param entityPlayer the player who opened the inventory.
+     */
+    @Override
+    public void closeInventory(final EntityPlayer entityPlayer)
+    {
+        /*
+         * This may be filled in order to specify some custom handling.
+         */
+    }
+
+    @Override
+    public boolean isItemValidForSlot(final int index, final ItemStack itemStack)
+    {
+        return index == 0 && itemStack != null && itemStack.getItem() instanceof ItemSeeds;
+    }
+
+    /**
+     * This may be used in order to return values of different GUI areas like the ones in the beacon.
+     * @param id the id of the field.
+     * @return the value of the field.
+     */
+    @Override
+    public int getField(int id)
+    {
+        return 0;
+    }
+
+    /**
+     * This may be used to set GUI areas with a certain id and value.
+     * @param id some id.
+     * @param value some value.
+     */
+    @Override
+    public void setField(int id, int value)
+    {
+        /*
+         * We currently need no fields.
+         */
+    }
+
+    /**
+     * Returns the number of fields.
+     * @return the amount.
+     */
+    @Override
+    public int getFieldCount()
+    {
+        return 0;
+    }
+
+    /**
+     * Completely clears the inventory.
+     */
+    @Override
+    public void clear()
+    {
+        for (int i = 0; i < this.stackResult.length; ++i)
+        {
+            this.stackResult[i] = null;
+        }
+    }
+    
+    /**
+     * Getter for the stack in the inventory. Since there is only one slot return always that one.
+     * @param index the slot.
+     * @return the itemStack in it.
+     */
+    @Override
+    public ItemStack getStackInSlot(int index)
+    {
+        return this.stackResult[0];
+    }
+
+    @Override
+    public ItemStack decrStackSize(int index, int count)
+    {
+        if (this.stackResult[index] != null)
+        {
+            if (this.stackResult[index].stackSize <= count)
+            {
+                ItemStack itemStack1 = this.stackResult[index];
+                this.stackResult[index] = null;
+                this.markDirty();
+                return itemStack1;
+            }
+            else
+            {
+                ItemStack itemstack = this.stackResult[index].splitStack(count);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'itemstack' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)